contracts/FindVerifier.cdc

```diff
@@ -0,0 +1,305 @@
+import FLOAT from "../contracts/standard/FLOAT.cdc"
+import FIND from "../contracts/FIND.cdc"
+import NonFungibleToken from "../contracts/standard/NonFungibleToken.cdc"
+import MetadataViews from "../contracts/standard/MetadataViews.cdc"
+
+pub contract FindVerifier {
+
+	pub struct interface Verifier {
+		pub let description : String
+		pub fun verify(_ param: {String : AnyStruct}) : Bool 
+	}
+
+	pub struct HasOneFLOAT : Verifier {
+		pub let floatEventIds : [UInt64] 
+		pub let description: String
+
+		init(_ floatEventIds: [UInt64]) {
+			pre{
+				floatEventIds.length > 0 : "list cannot be empty"
+			}
+			self.floatEventIds = floatEventIds
+			var desc = "User with one of these FLOATs are verified : "
+			for floatEventId in floatEventIds {
+				desc = desc.concat(floatEventId.toString()).concat(", ")
+			}
+			desc = desc.slice(from:0 ,upTo:desc.length-2)
+			self.description = desc
+		}
+
+		pub fun verify(_ param: {String : AnyStruct}) : Bool {
+
+			if self.floatEventIds.length == 0 {
+				return true
+			}
+
+			let user : Address = param["address"]! as! Address 
+			let float = getAccount(user).getCapability(FLOAT.FLOATCollectionPublicPath).borrow<&FLOAT.Collection{FLOAT.CollectionPublic}>() 
+
+			if float == nil {
+				return false
+			}
+
+			let floatsCollection=float!
+
+			let ids = floatsCollection.getIDs()
+			for id in ids {
+				let nft: &FLOAT.NFT = floatsCollection.borrowFLOAT(id: id)!
+				if self.floatEventIds.contains(nft.eventId) {
+					return true
+				}
+			}
+			return false
+		}
+
+	}
+
+	pub struct HasAllFLOAT : Verifier {
+		pub let floatEventIds : [UInt64] 
+		pub let description: String
+
+		init(_ floatEventIds : [UInt64]) {
+			pre{
+				floatEventIds.length > 0 : "list cannot be empty"
+			}
+			self.floatEventIds = floatEventIds
+			var desc = "User with all of these FLOATs are verified : "
+			for floatEventId in floatEventIds {
+				desc = desc.concat(floatEventId.toString()).concat(", ")
+			}
+			desc = desc.slice(from:0 ,upTo:desc.length-2)
+			self.description = desc
+		}
+
+		pub fun verify(_ param: {String : AnyStruct}) : Bool {
+
+			if self.floatEventIds.length == 0 {
+				return true
+			}
+
+			let user : Address = param["address"]! as! Address 
+			let float = getAccount(user).getCapability(FLOAT.FLOATCollectionPublicPath).borrow<&FLOAT.Collection{FLOAT.CollectionPublic}>() 
+
+			if float == nil {
+				return false
+			}
+
+			let floatsCollection=float!
+
+			let ids = floatsCollection.getIDs()
+			let checked : [UInt64] = []
+			for id in ids {
+				let nft: &FLOAT.NFT = floatsCollection.borrowFLOAT(id: id)!
+				if self.floatEventIds.contains(nft.eventId) && !checked.contains(nft.eventId) {
+					checked.append(nft.eventId)
+				}
+			}
+			
+			if checked.length == self.floatEventIds.length {
+				return true
+			}
+			return false
+		}
+	}
+
+	pub struct HasWhiteLabel : Verifier {
```

Comment:
I do not think WhiteLabel is the correct word here HasWhitelist? or IsInWhitelist?